runtime/test/buffer.cc

```diff
@@ -600,6 +600,30 @@ TEST(buffer, for_each_element) {
   ASSERT_TRUE(is_filled_buffer(buf, 7));
 }
 
+TEST(buffer, for_each_element_zero) {
+  // Verify that zero-dimensional buffers work
```

Comment:
This test wasn't failing before, right? Only with sanitizers?

I'd rather the test just test one buffer, so if it does fail for some reason, it's easier to debug.